types/next-auth/client.d.ts

```diff
@@ -48,18 +59,14 @@ declare function csrfToken(context?: NextContext): Promise<string | null>;
 declare const getCsrfToken: typeof csrfToken;
 declare function signin(
     provider?: string,
-    data?: GenericObject & {
-        callbackUrl?: string;
-        redirect?: boolean;
-    },
+    data?: SignInDataParams,
 ): Promise<void>;
+declare function signin(
+  provider: 'credentials',
+  data?: SignInDataParams,
+): Promise<CredentialsSignInReturn>;
```

Comment:
From what I can see [in the source code](https://github.com/nextauthjs/next-auth/blob/main/src/client/index.js#L250-L255), if all goes "well", `sigin()` always returns this shape:
```ts
interface singinSuccessPayload {
    error: string | undefined;
    status: number;
    ok: boolean;
    url: string | null;
}
```
no matter the type of provider specified 🧐 . I see, though, that in some cases, the promise [will resolve with `undefined`](https://github.com/nextauthjs/next-auth/blob/main/src/client/index.js#L208) before provoking a page redirect. Hence can we update the types to reflect this?
```ts
declare function signin(
  provider: 'string',
  data?: SignInDataParams,
): Promise<singinSuccessPayload | undefined>;
```

Reply:
Hey @lluia !! 
Thanks for your fast review.
I got about your suggestion but I would like to talk a little bit more because you said provider type not matter, but yes, they matter. 

[Like this line](https://github.com/nextauthjs/next-auth/blob/main/src/client/index.js#L235 where is verified if the provider is `credentials` or `email`.  [Check this line](https://github.com/nextauthjs/next-auth/blob/main/src/client/index.js#L212)

What do you think about it? 
Maybe could we include `email` in provider type?

Reply:
Cool point @guilouro 💅🏽 ,  maybe we can leave it as:
```ts
declare function signin(
  provider: 'credentials' | 'email' | 'string',
```
The ideal would be to have all the provider ids typed, but we don't have them yet 👍🏼

Reply:
Great!! I'll do these changes

Reply:
@lluia done! 🙂